spec/requests/agreements_spec.rb

```diff
@@ -1,53 +1,135 @@
+      end
+
+      it 'calls view agreements use-case and renders its response' do
+        get "/api/v1/agreements/#{tenancy_ref}"
+
+        expect(response.status).to eq(200)
+
+        parsed_response = JSON.parse(response.body)
+
+        expect(parsed_response['agreements'].count).to eq(1)
+        expect(parsed_response['agreements'].first['tenancyRef']).to eq(tenancy_ref)
+        expect(parsed_response['agreements'].first['agreementType']).to eq(agreement_type)
+        expect(parsed_response['agreements'].first['startingBalance']).to eq(starting_balance)
+        expect(parsed_response['agreements'].first['amount']).to eq(amount)
+        expect(parsed_response['agreements'].first['startDate']).to include(start_date.to_s)
+        expect(parsed_response['agreements'].first['frequency']).to eq(frequency)
+        expect(parsed_response['agreements'].first['currentState']).to eq(nil)
+        expect(parsed_response['agreements'].first['history']).to eq([])
+      end
+
+      it 'correctly maps all agreement_states in history' do
+        first_state = Hackney::Income::Models::AgreementState.create!(agreement_id: agreements_array.first.id, agreement_state: 'live')
+        second_state = Hackney::Income::Models::AgreementState.create!(agreement_id: agreements_array.first.id, agreement_state: 'breached')
+
+        get "/api/v1/agreements/#{tenancy_ref}"
+
+        expect(response.status).to eq(200)
 
-    before do
-      allow(Hackney::Income::ViewAgreements).to receive(:execute)
-        .with(tenancy_ref: tenancy_ref)
-        .and_return(agreements_response)
+        parsed_response = JSON.parse(response.body)
+
+        expect(parsed_response['agreements'].first['history']).to match([
+          {
+            'state' => first_state.agreement_state,
+            'date' => first_state.created_at.as_json
+          },
+          {
+            'state' => second_state.agreement_state,
+            'date' => second_state.created_at.as_json
+          }
+        ])
+      end
     end
+  end
+
+  describe 'POST /api/v1/agreement/{tenancy_ref}' do
+    path '/agreement/{tenancy_ref}' do
+      let(:create_agreement_instance) { instance_double(Hackney::Income::CreateAgreement) }
+      let(:tenancy_ref) { Faker::Number.number(digits: 2).to_s }
+      let(:agreement_type) { 'informal' }
+      let(:amount) { Faker::Commerce.price(range: 10...100) }
+      let(:start_date) { Faker::Date.between(from: 2.days.ago, to: Date.today) }
+      let(:frequency) { 'weekly' }
+      let(:current_state) { 'live' }
+      let(:starting_balance) { Faker::Commerce.price(range: 100...1000) }
+
+      let(:new_agreement_params) do
+        {
+          tenancy_ref: tenancy_ref,
+          agreement_type: agreement_type,
+          amount: amount.to_s,
+          start_date: start_date.to_s,
+          frequency: frequency
+        }
+      end
+
+      let(:created_agreement) do
+        Hackney::Income::Models::Agreement.create(
```

Comment:
Using the `create` method is a bit expensive because it actually hitting the DB, in this instance you could use `new` instead
```suggestion
        Hackney::Income::Models::Agreement.new(
```

At some point, we could set up [Factory bot](https://github.com/thoughtbot/factory_bot) and replace these to `create` method with `build` to stub them. Don't worry about this for now, we could come back to this when looking into speeding up the tests.

Reply:
I tried this because it does make sense, but found that it broke the second test for that endpoint - the one checking if states are mapped, because in the test, we create AgreementStates and you need an Agreement from the db I guess? 

So I've stuck to Create but yeah I guess it could be improved in the other ticket.